utils/exp_manager.py

```diff
@@ -747,7 +751,26 @@ def hyperparameters_optimization(self) -> None:
         )
 
         try:
-            study.optimize(self.objective, n_trials=self.n_trials, n_jobs=self.n_jobs)
+            if self.total_n_trials is not None:
+                counted_states = [
+                    TrialState.COMPLETE,
+                    TrialState.RUNNING,
+                    TrialState.PRUNED,
+                ]
+                completed_trials = sum(t.state in counted_states for t in study.trials)
```

Comment:
Optuna has an API to get all trials by trial state:
```suggestion
                completed_trials = len(study.get_trials(state= counted_states))
```